lib/livebook_web/live/apps_live.ex

```diff
@@ -42,79 +42,264 @@ defmodule LivebookWeb.AppsLive do
+                      value={@selected_app_folder}
+                      options={@app_folder_options}
+                    />
+                  </form>
+                </div>
+              </div>
             </div>
-          </div>
-          <div
-            :if={@apps == [] and not @empty_apps_path?}
-            class="mt-5 flex flex-col w-full max-w-[400px]"
-          >
-            <.no_entries :if={@apps == []}>
-              No apps running.
-            </.no_entries>
-          </div>
-          <div :if={@apps == [] and @empty_apps_path?} class="mt-5 text-gray-600">
-            <div>
-              No app notebooks found. Follow these steps to list your apps here:
+
+            <div
+              :if={@filtered_apps == []}
+              class="bg-white rounded-lg shadow-sm border border-gray-200 p-12 text-center"
+            >
+              <.remix_icon icon="windy-line" class="text-gray-400 text-2xl" />
+              <h3 class="text-lg font-medium text-gray-900">No apps found</h3>
+              <p class="text-gray-600">Try adjusting your search or filter criteria</p>
+            </div>
+            <div
+              :for={{app_folder, id, icon, apps} <- @grouped_apps}
+              :if={@filtered_apps != []}
+              id={id}
+              class="mb-8"
+            >
+              <h2 class="text-xl font-semibold text-gray-900 mb-4 flex items-center">
+                <.remix_icon icon={icon} class="mr-2" />
+                {app_folder}
+                <span class="ml-2 text-sm font-normal text-gray-500">({length(apps)})</span>
+              </h2>
+              <div class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-4">
+                <.link
+                  :for={app <- apps_listing(apps)}
+                  id={"app-#{app.slug}"}
+                  navigate={~p"/apps/#{app.slug}"}
+                  class="border bg-gray-50 border-gray-300 rounded-lg p-4 hover:shadow-md hover:border-blue-300 transition-all duration-200"
+                >
+                  <div class="flex items-center justify-between">
+                    <h3 class="text-sm font-medium text-gray-900 group-hover:text-blue-600 transition-colors truncate">
+                      {app.notebook_name}
+                    </h3>
+                    <div class="space-x-1 ml-2">
+                      <.remix_icon
+                        :if={not app.public?}
+                        icon="lock-password-line"
+                        class="h-4 w-4 text-gray-400"
+                      />
+                      <.remix_icon
+                        icon="arrow-right-line"
+                        class="h-4 w-4 text-gray-400 group-hover:text-blue-600 transition-colors"
+                      />
+                    </div>
+                  </div>
+                </.link>
+              </div>
+            </div>
+          <% else %>
+            <div
+              :if={@empty_apps_path?}
```

Comment:
Please also post a screenshot of how this particular screen looks like (you can manually swap the conditionals in the code).